Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5100,9 +5100,15 @@ bool RenderBox::establishesIndependentFormattingContext() const
     return isGridItem() || RenderElement::establishesIndependentFormattingContext();
 }
 
+static bool isReplacedElement(Element* element)
+{
+    // Checkboxes and radioboxes are not isReplaced() nor do they have their own renderer in which to override avoidFloats().
+    return element && element->isElementNode() && element->isFormControlElement();
+}
+
 bool RenderBox::avoidsFloats() const
 {
-    return isReplacedOrInlineBlock() || isLegend() || isFieldset() || createsNewFormattingContext();
+    return isReplacedOrInlineBlock() || isReplacedElement(element()) || isLegend() || isFieldset() || createsNewFormattingContext();
```

Comment:
Shouldn't this be part of isReplacedOrInlineBlock()?

Reply:
It seems it isn't.

Reply:
@alanbaradlay - (1) and (4) looks like good. Should I do those and for (2) and (3) - do follow-up bug?

Reply:
yeah, #1 and #4 should do.

Reply:
@alanbaradlay - Added your explanation from (1) in commit and slightly updated title and also pushed changes suggested in (4). Thanks a lot for guidance and input. @annevk - thanks for your input as well. Always good to learn something especially in Layout & Rendering code.